### ISSUE TITLE
RFE: enable rootless for ldap users?

### ISSUE THREAD

**Issue by nimbius** [containers/podman#5196]
/kind feature

**Description**

It would be awesome to have rootless pods available for LDAP users.  currently this appearsr to be a subuid/subgid limitation?
https://github.com/containers/libpod/blob/master/docs/tutorials/rootless_tutorial.md

> These files must be available locally to the host machine. It is not possible to configure this with LDAP or Active Directory.

**Steps to reproduce the issue:**

1.any podman/buildah/skopeo operation as a user

**Describe the results you received:**
tar failures, lchown failures.  every dang failure. :(

**Describe the results you expected:**
pods pods pods! :)

**Additional information you deem important (e.g. issue happens only occasionally):**

bash-4.4$ podman version
Version:            1.6.4
RemoteAPI Version:  1
Go Version:         go1.12.12
OS/Arch:            linux/amd64

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(bash-4.4$ rpm -qi podman
Name        : podman
Version     : 1.6.4
Release     : 2.module+el8.1.1+5363+bf8ff1af
Architecture: x86_64
Install Date: Tue 11 Feb 2020 09:00:48 AM PST
Group       : Unspecified
Size        : 59988453
License     : ASL 2.0
Signature   : RSA/SHA256, Mon 06 Jan 2020 06:57:25 AM PST, Key ID 199e2f91fd431d51
Source RPM  : podman-1.6.4-2.module+el8.1.1+5363+bf8ff1af.src.rpm
Build Date  : Fri 03 Jan 2020 04:20:53 AM PST
Build Host  : x86-vm-09.build.eng.bos.redhat.com
Relocations : (not relocatable)
Packager    : Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>
Vendor      : Red Hat, Inc.
URL         : https://podman.io/
Summary     : Manage Pods, Containers and Container Images

```
tested in RHEL8, with SSSD, podman, buildah and skopeo

**Comment by mheon** [comment]
The core of the issue is outside of Podman - we'd need support in `newuidmap` and `newgidmap` for this. The issue below has some details:

https://github.com/shadow-maint/shadow/issues/154

**Comment by vrothberg** [comment]
Thanks for opening the issue, @nimbius. As we're blocked on shadow-maint/shadow#154 there's not much we can do atm. Given it's documented, I am closing the issue.

Once shadow-maint/shadow#154 is fixed and has bubbled up, we can tackle it here in libpod.

**Comment by d-a-v** [comment]
https://github.com/shadow-maint/shadow/issues/154 's title is `/etc/subuid /etc/subgid available via the network #154`.

Before noticing this dependency, I have tried to *create local `sub?id` files*.
Say my ldap uid is 9876:
```
# cat /etc/sub[ug]id
gauchard:987600000:99999
gauchard:987600000:99999
#32b:   4294967296  uid*100000 may allow uid up to 40000
```
and I still get
```
ERRO[0005] Error while applying layer: ApplyLayer exit status 1 stdout:  stderr: there might not be enough IDs available in the namespace (requested 0:42 for /etc/shadow): lchown /etc/shadow: invalid argument 
```
I also tried to add/copy from ldap my entry into `/etc/passwd` and `/etc/shadow`. 
Would this method work (and I have then to look for my issue elsewhere) ?
or is it just not possible for a user whose uid is also a valid ldap uid ?

Just in case:
```
# sysctl kernel.unprivileged_userns_clone
kernel.unprivileged_userns_clone = 1

$ rm -rf ~/.config/containers ~/.local/share/containers
$ podman system migrate
$ podman unshare cat /proc/self/uid_map
         0       9876          1
         1  987600000      99999

```
(no issue with `root` user)

**Comment by rhatdan** [comment]
IN the unshare could you attempt to

```
# touch test
# chown 0:42 test
```

**Comment by d-a-v** [comment]
Thanks for the answer !
When I came back to my issue some hours ago and after tweaking around (I created a local-only user for testing purpose), the issue with the ldap account disappeared without warning.
So I came back here and removed my message above because it is following a closed issue with an identified upstream cause. It sounds like I didn't succeed in deleting it as I believed.
I now understand the error message thanks to your answer.
```
$ ls -aln /etc/shadow
-rw-r----- 1 0 42 976 Jun 29 10:57 /etc/shadow
```
So the explanation *might be* that I broke the system the other day, and some sanity script repaired "permissions" in between.

podman is working with ldap users, and that's great !

**Comment by TomSweeneyRedHat** [comment]
@espindola thanks for the pointer/reminder on shadow-maint.  Reopening.

**Comment by rhatdan** [comment]
@giuseppe opened a PR for this
https://github.com/containers/storage/pull/882
But there has been no release of shadow-utils yet, so distributions are not shipping libsubid yet.

**Comment by rhatdan** [comment]
This should now be available in rawhide.  Containers/storage has been merged into main.
